tests/models/clip/test_clip.py

```diff
@@ -0,0 +1,21 @@
+from PIL import Image
+import requests
+
+from transformers import CLIPProcessor, CLIPModel
```

Comment:
need to add them to requirements-dev.txt?

Reply:
Indeed.
* `from PIL import Image`: Needs package `Pillow`, which has existed in `requirements-dev.txt`.
* `import requests`: Needs package `requests`, which requires update of `requirements-dev.txt`. The update was done.
* `from transformers import CLIPProcessor, CLIPModel`: Needs package `transformers`, which has existed in `requirements-dev.txt`.

Reply:
@cthsieh 
```
ERROR: Cannot install -r requirements-dev.txt (line 5), jupyterlab, metal-libs, requests==2.22.0 and transformers because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested requests==2.22.0
    transformers 4.38.0 depends on requests
    dash 2.15.0 depends on requests
    huggingface-hub 0.19.3 depends on requests
    jupyterlab-server 2.27.3 depends on requests>=2.31
    The user requested requests==2.22.0
    transformers 4.38.0 depends on requests
    dash 2.15.0 depends on requests
    huggingface-hub 0.19.3 depends on requests
    jupyterlab-server 2.27.2 depends on requests>=2.31
    The user requested requests==2.22.0
    transformers 4.38.0 depends on requests
    dash 2.15.0 depends on requests
    huggingface-hub 0.19.3 depends on requests
    jupyterlab-server 2.27.1 depends on requests>=2.31

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip to attempt to solve the dependency conflict
```

see details here https://github.com/tenstorrent/pytorch2.0_ttnn/actions/runs/10518866320/job/29145318003

Reply:
ERROR at teardown of test_clip :monocle_face:

What decides the page size anyway?  Not even divisible by 4?

```
Page size must be divisible by sizeof(uint32_t) because buffers hold uint32_t values
CLIP compiled failed to run. Raised exception: TT_FATAL @ ../tt_metal/impl/buffers/buffer.cpp:41: page_size % sizeof(uint32_t) == 0
info:
Page size must be divisible by sizeof(uint32_t) because buffers hold uint32_t values
backtrace:
 --- tt::tt_metal::validate_buffer_size_and_page_size(unsigned long, unsigned long, tt::tt_metal::BufferType const&, tt::tt_metal::TensorMemoryLayout const&, std::__1::optional<tt::tt_metal::ShardSpecBuffer> const&)
 --- tt::tt_metal::Buffer::Buffer(tt::tt_metal::Device*, unsigned long, unsigned long, tt::tt_metal::BufferType, tt::tt_metal::TensorMemoryLayout, std::__1::optional<tt::tt_metal::ShardSpecBuffer> const&, std::__1::optional<bool>, bool)
 --- /home/ubuntu/jdh8/tt-metal/ttnn/ttnn/_ttnn.so(+0xc93e8e) [0x7fef32502e8e]
 --- tt::tt_metal::tensor_impl::allocate_buffer_on_device(unsigned int, tt::tt_metal::Device*, tt::tt_metal::Shape const&, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig const&, std::__1::optional<tt::tt_metal::ShardSpecBuffer> const&)
 --- tt::tt_metal::create_device_tensor(tt::tt_metal::Shape const&, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::Device*, tt::tt_metal::MemoryConfig const&)
 --- /home/ubuntu/jdh8/tt-metal/ttnn/ttnn/_ttnn.so(_ZN2tt8tt_metal9operation29generic_create_output_tensorsIN4ttnn10operations13data_movement21UntilizeWithUnpaddingEEENS1_21program_output_helperIT_Xsr18has_create_programIS8_EE5valueEE4typeERKS8_RKNSt3__16vectorINS0_6TensorENSD_9allocatorISF_EEEENSD_8optionalINS0_8DataTypeEEENS0_6LayoutERKNSL_INS0_12MemoryConfigEEE+0x178) [0x7fef322c1638]
 --- ttnn::operations::data_movement::UntilizeWithUnpadding::create_output_tensors(std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>> const&, std::__1::vector<std::__1::optional<tt::tt_metal::Tensor>, std::__1::allocator<std::__1::optional<tt::tt_metal::Tensor>>> const&) const
 --- /home/ubuntu/jdh8/tt-metal/ttnn/ttnn/_ttnn.so(+0xa4f9b9) [0x7fef322be9b9]
 --- tt::tt_metal::operation::detail::OldInfraDeviceOperation<std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>>>::tensor_return_value_t ttnn::device_operation::detail::launch_on_single_device<tt::tt_metal::operation::detail::OldInfraDeviceOperation<std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>>>>(unsigned char, tt::tt_metal::operation::detail::OldInfraDeviceOperation<std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>>>::operation_attributes_t const&, tt::tt_metal::operation::detail::OldInfraDeviceOperation<std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>>>::tensor_args_t const&)
 --- /home/ubuntu/jdh8/tt-metal/ttnn/ttnn/_ttnn.so(+0x2bc4fa) [0x7fef31b2b4fa]
 ```

Reply:
Take a look here:

https://github.com/tenstorrent/tt-metal/blob/7e56ce1c94ff8522f0a9e49f31769ec592fba366/ttnn/cpp/ttnn/tensor/tensor_impl.cpp#L46

I do have a pending PR that adds some lowering conditions for when the page size doesn't meet the restrictions. This PR might not fix this particular issue, but you can probably add a condition similar to this. https://github.com/tenstorrent/pytorch2.0_ttnn/blob/0a5e072e6b02865cd47eb25f821322f598f74cdf/torch_ttnn/passes/lowering/to_tt_pass.py#L151-L170